source/developers/cook-books/aws/upload-transcode-video.rst

```diff
@@ -0,0 +1,239 @@
+
+.. code-block:: xml
+   :linenos:
+
+       <?xml version="1.0" encoding="UTF-8"?>
+       <aws>
+         <s3>
+           <profile>
+             <id>s3-default</id>
+             <credentials>
+               <accessKey>XXXXXXXXXX</accessKey>
+               <secretKey>XXXXXXXXXXXXXXXXXXXX</secretKey>
+             </credentials>
+             <region>us-east-1</region>
+             <bucketName>mybucket</bucketName>
+           </profile>
+         </s3>
+
+         <mediaConvert>
+           <profile>
+             <id>mediaconvert-default</id>
+             <credentials>
+               <accessKey>xxxxxxxxx</accessKey>
+               <secretKey>xxxxxxxxx</secretKey>
+             </credentials>
+             <region>us-east-1</region>
+             <endpoint>https://XXXXXXXX.mediaconvert.us-east-1.amazonaws.com</endpoint>
+             <role>arn:aws:iam::XXXXXXXXXXXX:role/...</role>
+             <queue>arn:aws:mediaconvert:us-east-1:XXXXXXXXXXXX:queues/...</queue>
+             <inputPath>mybucket/folder/videos</inputPath>
+             <template>Example Template</template>
+           </profile>
+         </mediaConvert>
+       </aws>
+
+|
+
+-------------------------------------------------------------------------------------------
+Step 2: Add the data source for uploading video to AWS for transcoding, to the content type
+-------------------------------------------------------------------------------------------
+
+For our example, we'll add the Video Transcoding from S3 datasource and a Video picker control to the ``Page - Article`` content type.  To do this:
+
+#. Go to |siteConfig| -> **Content Types** and open the ``Page - Article`` content type definition.
+#. We'll add a data source where to upload a video that will be transcoded.  At the end of the **Data Sources** section, add a ``Video Transcoding from S3`` data source with Title **MediaConvert Transcode** and Name **mediaConvertTranscode**.  In the **Input Profile Id** property, enter the MediaConvert ``profile.id`` configured in step 1, used for uploading and triggering the transcode job (``mediaconvert-default`` from the example in step 1).  In the **Output Profile Id** property, enter the S3 ``profile.id`` configured in step 1, used to access the files generated by the transcode job (``s3-default`` from the example in step 1).
+
+   .. image:: /_static/images/guides/s3/mediaConvert-datasource.png
+      :alt: AWS MediaConvert  - Video Transcoding from S3 Data Sources
+      :align: center
+      :width: 65%
+
+   |
+
+#. At the end of the **Content** section, add a ``Video`` control with Title **Video** and Name
+   **video**.
+
+   .. image:: /_static/images/guides/s3/mediaConvert-video-picker.png
+      :alt: AWS MediaConvert - Video picker
+      :align: center
+      :width: 65%
+
+   |
+
+
+------------------------------------------------
+Step 3: Add Freemarker code to render the videos
```

Comment:
Maybe a note to those not using FreeMarker. Or just say `if you're using FreeMarker then do this`?

Reply:
I guess the freemarker step is there since we are using the website editorial for our example on using the AWS datasource.

Should I make it a note? Also, if we make that a note, should I change the one in: https://docs.craftercms.org/en/3.1/developers/cook-books/aws/use-s3-to-store-assets.html?
 Thanks!

Reply:
Yes, we do need a note for both. Something like "If you're using FreeMarker as your view layer, follow these steps to ..."